### PR TITLE
[Ragile]: Add new centec platform ra-b6010 

### PR DESCRIPTION
**Pull request by pettershao-ragilenetworks**
#### What I did it
Add new platform  arm64-ragile_ra-b6010-48gt4x-r0 (Centec)
ASIC Vendor: Centec
Switch ASIC: Centec
Port Config: 48x1G+4x10G

#### Why I did it
Add new platform RA-B6010-48GT4X
#### How I did it
Add new platform RA-B6010-48GT4X
#### How to verify it
show version
docker ps -a
show platform syseeprom
show platform fan
show platform psustatus
show platform temperature
show platform firmware status
show platform ssdhealth
show interface status
show interface transceiver presence
show interface transceiver eeprom
show environment
sensors
sudo show system-health detail
#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211

#### A picture of a cute animal (not mandatory but encouraged)

### CODE REVIEW

**Comment by Stephengzh-Ragilenetworks** [comment]
Platform basic tests:

show version
docker ps -a
show platform syseeprom
show platform fan
show platform psustatus
show platform temperature
show platform firmware status

show platform ssdhealth
show interface status

show interface transceiver presence
show interface transceiver eeprom
show environment

**Comment by pettershao-ragilenetworks** [comment]
@lguohan @yxieca help check the test result, thanks!

**Comment by pettershao-ragilenetworks** [comment]
[report_6010.zip](https://github.com/sonic-net/sonic-buildimage/files/11953094/report_6010.zip)
@lguohan @yxieca newest pytest report with version include platform.json, help check and forward, thanks!

File: platform/centec-arm64/sonic-platform-modules-ragile/ra-b6010-48gt4x/sonic_platform/sfp.py [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+        return transceiver_info
+
+    def set_optoe_write_max(self, write_max):
+        """
+        This func is declared and implemented by SONiC but we're not supported
+        so override it as NotImplemented
+        """
+        self._sfplog(LOG_DEBUG_LEVEL, "set_optoe_write_max NotImplemented")
+
+    def refresh_xcvr_api(self):
```
[the remark this answers is not in the document]
Comment (philo-micas) [reply platform/centec-arm64/sonic-platform-modules-ragile/ra-b6010-48gt4x/sonic_platform/sfp.py:100]: > @pettershao-ragilenetworks is not called during optics removal followed by insertion?

No, will not be called.

File: platform/centec-arm64/sonic-platform-modules-ragile/ra-b6010-48gt4x/sonic_platform/sfp.py [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+            optoe_type = self.OPTOE_DRV_TYPE3
+        elif 'Sff8472Api' in class_name:
+            self.sfp_type = 'SFP'
+            optoe_type = self.OPTOE_DRV_TYPE2
+        elif ('Sff8636Api' in class_name or 'Sff8436Api' in class_name):
+            self.sfp_type = 'QSFP'
+            optoe_type = self.OPTOE_DRV_TYPE1
+        # set optoe driver
+        if optoe_type is not None:
+            self._sfp_api.set_optoe_type(optoe_type)
```
[the remark this answers is not in the document]
Comment (philo-micas) [reply platform/centec-arm64/sonic-platform-modules-ragile/ra-b6010-48gt4x/sonic_platform/sfp.py:119]: @prgeor  Yes, they can in base class, but keep in mind that they are just partial.

**Review by prgeor — APPROVED** [review]

**Review by prgeor — COMMENTED** [review]
@yxieca help merge

**Comment by lguohan** [comment]
@pettershao-ragilenetworks , this pr is reverted, because it has duplicated optoe driver which is already in sonic-linux-kernel. also, it has duplicated driver for pca954x, pca9641, at24, and etc. on.

We should use original kernel driver. if original driver cannot be used, we need to reason.

**Pull request merged**